src/hdate.js

```diff
@@ -287,18 +287,37 @@ prototype[abs] = function() {
 };
 
 prototype.toString = function(o) {
-	return c.LANG([this[getDate](), null, gematriya(this[getDate]())], o) + ' ' +
+	return this.getDayName(o) + ' ' +
 		this.getMonthName(o) + ' ' +
-		c.LANG([this[getFullYear](), null, gematriya(this[getFullYear]())], o);
+		this.getYearName(o);
+};
+
+prototype.getWeekDayName = function(o) {
+	let dayObj = c.daysNames[this.getDay()];
+	if (dayObj){
+		return c.daysNames[this.getDay()][o] ? c.daysNames[this.getDay()][o] : c.daysNames[this.getDay()]['eng'];
```

Comment:
(Been a while since i used this codebase but) pretty sure you should use c.LANG() for parsing the language option instead of accessing it directly from the object. Unless you have a good reason for doing it differently here, which should be mentioned in a comment.